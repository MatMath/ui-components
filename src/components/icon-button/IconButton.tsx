import * as React from 'react';
import { getClassNames } from '@utility/cssUtils';
import { Button } from '@components/button/Button';
import { ButtonSize, ButtonTheme } from '@components/button/types';
import styles from './IconButton.module.scss';

interface IconButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  icon: JSX.Element;
  size?: ButtonSize;
  theme?: ButtonTheme;
}

export const IconButton: React.FC<IconButtonProps> = ({
  size = 'small',
  theme = 'light',
  icon,
  ...htmlButtonProps
}) => {
  return (
    <Button
      nature={'ghost'}
      size={size}
      theme={theme}
      {...htmlButtonProps}
      className={getClassNames(styles.iconButton, styles[size])}
    >
      {icon}
    </Button>
  );
};
