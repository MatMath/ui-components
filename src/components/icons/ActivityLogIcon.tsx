import * as React from 'react';
import { IconProps } from '@components/icons/types';
import { SvgWrapper } from '@components/icons/utils/SvgWrapper';

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */

export const ActivityLogIcon: React.FC<IconProps> = (props: IconProps) => (
  <SvgWrapper {...props}>
    <React.Fragment>
      <path d='M4 1.50016H2.5V18.5002H14.2L13.3 17.0002H4V1.50016Z' />
      <path d='M16 17.0002L16 18.5002H17.5V17.0002H16Z' />
      <path
        fillRule='evenodd'
        clipRule='evenodd'
        d='M16 6.31083L16 18.5002H17.5L17.5 5.68951L13.3107 1.50016L2.5 1.50016L2.5 3.00016L12.6893 3.00016L16 6.31083Z'
      />
      <path
        fillRule='evenodd'
        clipRule='evenodd'
        d='M11.8093 1.5H13.3093L13.3093 5.68328L17.5 5.68342L17.5 7.18342L11.8093 7.18323L11.8093 1.5Z'
      />
      <path d='M8.5 10.5L13.6016 10.4997L14.5 9.00097L8.5 9V10.5Z' />
      <path d='M13.5996 13.7477L8.5 13.75V12.25H14.4975L13.5996 13.7477Z' />
      <path d='M5.5 9H7V10.5H5.5V9Z' />
      <path d='M7 12.25H5.5V13.75H7V12.25Z' />
    </React.Fragment>
  </SvgWrapper>
);

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */
