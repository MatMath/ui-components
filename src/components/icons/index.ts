/*-- EXPORT ICON COMPONENTS --*/
export { WindowsIcon } from './WindowsIcon';
export { WebIcon } from './WebIcon';
export { WarningIcon } from './WarningIcon';
export { UserCircleIcon } from './UserCircleIcon';
export { VaultIcon } from './VaultIcon';
export { UserCircleIcon } from './UserCircleIcon';
export { UnlockIcon } from './UnlockIcon';
export { ToolsIcon } from './ToolsIcon';
export { TaxIcon } from './TaxIcon';
export { ShieldFlagIcon } from './ShieldFlagIcon';
export { ShieldCheckIcon } from './ShieldCheckIcon';
export { SharingGroupIcon } from './SharingGroupIcon';
export { SharedIcon } from './SharedIcon';
export { ShareIcon } from './ShareIcon';
export { SettingsIcon } from './SettingsIcon';
export { SecurityDashboardIcon } from './SecurityDashboardIcon';
export { SearchIcon } from './SearchIcon';
export { RevokeIcon } from './RevokeIcon';
export { RevealIcon } from './RevealIcon';
export { RefreshIcon } from './RefreshIcon';
export { PhoneIcon } from './PhoneIcon';
export { PersonalInfoIcon } from './PersonalInfoIcon';
export { PaymentsIcon } from './PaymentsIcon';
export { PasswordsIcon } from './PasswordsIcon';
export { PasswordHistoryIcon } from './PasswordHistoryIcon';
export { PasswordChangerIcon } from './PasswordChangerIcon';
export { OpenWebsiteIcon } from './OpenWebsiteIcon';
export { NoNetworkIcon } from './NoNetworkIcon';
export { MoreIcon } from './MoreIcon';
export { MobileIcon } from './MobileIcon';
export { MailIcon } from './MailIcon';
export { LogOutIcon } from './LogOutIcon';
export { LockIcon } from './LockIcon';
export { LoaderIcon } from './LoaderIcon';
export { LinkIcon } from './LinkIcon';
export { InfoCircleIcon } from './InfoCircleIcon';
export { IdIcon } from './IdIcon';
export { HouseIcon } from './HouseIcon';
export { HideIcon } from './HideIcon';
export { HelpIcon } from './HelpIcon';
export { GeneratorIcon } from './GeneratorIcon';
export { ForwardIcon } from './ForwardIcon';
export { EditIcon } from './EditIcon';
export { DotsIcon } from './DotsIcon';
export { DashlaneLogoIcon } from './DashlaneLogoIcon';
export { DashboardIcon } from './DashboardIcon';
export { CrossCircleIcon } from './CrossCircleIcon';
export { CopyPasswordIcon } from './CopyPasswordIcon';
export { CopyIcon } from './CopyIcon';
export { ComputerIcon } from './ComputerIcon';
export { CloseIcon } from './CloseIcon';
export { CheckCircleIcon } from './CheckCircleIcon';
export { CheckIcon } from './CheckIcon';
export { CaretDownIcon } from './CaretDownIcon';
export { BuildingIcon } from './BuildingIcon';
export { BellIcon } from './BellIcon';
export { BackIcon } from './BackIcon';
export { ArrowDownIcon } from './ArrowDownIcon';
export { AppleIcon } from './AppleIcon';
export { AndroidIcon } from './AndroidIcon';
export { AddIcon } from './AddIcon';
export { ActivityLogIcon } from './ActivityLogIcon';
export { AddIcon } from './AddIcon';
export { AndroidIcon } from './AndroidIcon';
export { AppleIcon } from './AppleIcon';
export { ArrowDownIcon } from './ArrowDownIcon';
export { BackIcon } from './BackIcon';
export { BellIcon } from './BellIcon';
export { BuildingIcon } from './BuildingIcon';
export { CaretDownIcon } from './CaretDownIcon';
export { CheckIcon } from './CheckIcon';
export { CheckCircleIcon } from './CheckCircleIcon';
export { CloseIcon } from './CloseIcon';
export { ComputerIcon } from './ComputerIcon';
export { CopyIcon } from './CopyIcon';
export { CopyPasswordIcon } from './CopyPasswordIcon';
export { CrossCircleIcon } from './CrossCircleIcon';
export { DashboardIcon } from './DashboardIcon';
export { DashlaneLogoIcon } from './DashlaneLogoIcon';
export { DotsIcon } from './DotsIcon';
export { EditIcon } from './EditIcon';
export { ForwardIcon } from './ForwardIcon';
export { GeneratorIcon } from './GeneratorIcon';
export { HelpIcon } from './HelpIcon';
export { HideIcon } from './HideIcon';
export { HouseIcon } from './HouseIcon';
export { IdIcon } from './IdIcon';
export { InfoCircleIcon } from './InfoCircleIcon';
export { LinkIcon } from './LinkIcon';
export { LoaderIcon } from './LoaderIcon';
export { LockIcon } from './LockIcon';
export { LogOutIcon } from './LogOutIcon';
export { MailIcon } from './MailIcon';
export { MobileIcon } from './MobileIcon';
export { MoreIcon } from './MoreIcon';
export { NoNetworkIcon } from './NoNetworkIcon';
export { OpenWebsiteIcon } from './OpenWebsiteIcon';
export { PasswordChangerIcon } from './PasswordChangerIcon';
export { PasswordHistoryIcon } from './PasswordHistoryIcon';
export { PasswordsIcon } from './PasswordsIcon';
export { PaymentsIcon } from './PaymentsIcon';
export { PersonalInfoIcon } from './PersonalInfoIcon';
export { PhoneIcon } from './PhoneIcon';
export { RefreshIcon } from './RefreshIcon';
export { RevealIcon } from './RevealIcon';
export { RevokeIcon } from './RevokeIcon';
export { SearchIcon } from './SearchIcon';
export { SecurityDashboardIcon } from './SecurityDashboardIcon';
export { SettingsIcon } from './SettingsIcon';
export { ShareIcon } from './ShareIcon';
export { SharedIcon } from './SharedIcon';
export { SharingGroupIcon } from './SharingGroupIcon';
export { ShieldCheckIcon } from './ShieldCheckIcon';
export { ShieldFlagIcon } from './ShieldFlagIcon';
export { TaxIcon } from './TaxIcon';
export { ToolsIcon } from './ToolsIcon';
export { UnlockIcon } from './UnlockIcon';
export { VaultIcon } from './VaultIcon';
export { UserCircleIcon } from './UserCircleIcon';
export { WarningIcon } from './WarningIcon';
export { WebIcon } from './WebIcon';
export { WindowsIcon } from './WindowsIcon';
