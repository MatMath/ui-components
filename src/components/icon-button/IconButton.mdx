---
name: IconButton
menu: Components
---

import { Playground, Props } from 'docz';
import { IconButton } from './IconButton';
import { CopyPasswordIcon, OpenWebsiteIcon, MoreIcon } from '../icons';
import { white, midGreen00, dashGreen00 } from '../../colors/colors.ts';
import '../../utility/playgroundStyles.css';

# IconButton

A wrapper of the Button component but that holds a single icon.

## Properties

<Props of={IconButton} />

## Usage

<Playground className={{preview: 'playground-base'}}>
    <IconButton icon={<MoreIcon/>} />

    <IconButton icon={<OpenWebsiteIcon/>} />

    <IconButton icon={<CopyPasswordIcon/>} />

</Playground>

### Theme

The default theme is `light` (white), but some buttons can be displayed in a `dark` theme (Dashlane's dash green, not an actual _dark_ theme).

<Playground className={{preview: 'playground-base'}} style={{ preview: {backgroundColor: dashGreen00} }}>
    <IconButton theme='dark' icon={<MoreIcon color={white}/>} />

    <IconButton theme='dark' icon={<OpenWebsiteIcon color={white}/>} />

    <IconButton theme='dark' icon={<CopyPasswordIcon color={white}/>} />

</Playground>

### Size

Buttons can be have different sizes [`small`, `medium`, `large`] with `small` being the default value if none is specified.

<Playground className={{preview: 'playground-base'}} style={{ preview: {minHeight: '150px'} }}>

    <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'space-between' }}>
      <IconButton icon={<MoreIcon/>} size='small'/>
      <IconButton icon={<MoreIcon/>} size='medium'/>
      <IconButton icon={<MoreIcon/>} size='large'/>
    </div>

    <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'space-between' }}>
      <IconButton icon={<OpenWebsiteIcon/>} size='small'/>
      <IconButton icon={<OpenWebsiteIcon/>} size='medium'/>
      <IconButton icon={<OpenWebsiteIcon/>} size='large'/>
    </div>

    <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'space-between' }}>
      <IconButton icon={<CopyPasswordIcon/>} size='small'/>
      <IconButton icon={<CopyPasswordIcon/>} size='medium'/>
      <IconButton icon={<CopyPasswordIcon/>} size='large'/>
    </div>

</Playground>

### Disabled

In disabled pseudo-state, buttons behave as following.

<Playground className={{preview: 'playground-base'}}>

    <IconButton disabled icon={<MoreIcon/>} />

    <IconButton disabled icon={<OpenWebsiteIcon/>} />

    <IconButton disabled icon={<CopyPasswordIcon/>} />

</Playground>
