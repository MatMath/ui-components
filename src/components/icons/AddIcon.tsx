import * as React from 'react';
import { IconProps } from '@components/icons/types';
import { SvgWrapper } from '@components/icons/utils/SvgWrapper';

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */

export const AddIcon: React.FC<IconProps> = (props: IconProps) => (
  <SvgWrapper {...props}>
    <React.Fragment>
      <path d='M9.25 2H10.75V18H9.25V2Z' />
      <path d='M2 11V9.5H18V11H2Z' />
    </React.Fragment>
  </SvgWrapper>
);

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */
