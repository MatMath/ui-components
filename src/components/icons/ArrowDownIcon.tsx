import * as React from 'react';
import { IconProps } from '@components/icons/types';
import { SvgWrapper } from '@components/icons/utils/SvgWrapper';

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */

export const ArrowDownIcon: React.FC<IconProps> = (props: IconProps) => (
  <SvgWrapper {...props}>
    <React.Fragment>
      <path
        fillRule='evenodd'
        clipRule='evenodd'
        d='M10 11.5503L5.63659 6L4.5 7.00399L10 14L15.5 7.00399L14.3634 6L10 11.5503Z'
      />
    </React.Fragment>
  </SvgWrapper>
);

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */
