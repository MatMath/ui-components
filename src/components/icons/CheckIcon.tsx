import * as React from 'react';
import { IconProps } from '@components/icons/types';
import { SvgWrapper } from '@components/icons/utils/SvgWrapper';

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */

export const CheckIcon: React.FC<IconProps> = (props: IconProps) => (
  <SvgWrapper {...props}>
    <React.Fragment>
      <path
        fillRule='evenodd'
        clipRule='evenodd'
        d='M18 3.93779L7.89681 16L2 11.0719L2.89069 9.95575L7.72252 13.9939L16.9309 3L18 3.93779Z'
      />
    </React.Fragment>
  </SvgWrapper>
);

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */
