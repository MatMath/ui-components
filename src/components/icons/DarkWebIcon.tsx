import * as React from 'react';
import { IconProps } from '@components/icons/types';
import { SvgWrapper } from '@components/icons/utils/SvgWrapper';

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */

export const DarkWebIcon: React.FC<IconProps> = (props: IconProps) => (
  <SvgWrapper {...props}>
    <React.Fragment>
      <path
        fillRule='evenodd'
        clipRule='evenodd'
        d='M12.3252 16.2632C11.8418 16.5244 11.3469 16.8102 10.8749 17.0928L10.8749 11.7311L15.9402 14.6375C15.5005 14.8107 15.0409 14.9969 14.5909 15.1876L15.5988 16.9333C17.4602 16.1565 19.5 15.445 19.5 15.445C19.5 15.445 19.1042 12.123 19.1042 9.99446C19.1042 7.8659 19.5 4.54394 19.5 4.54394C19.5 4.54394 15.9679 3.31195 13.9583 2.32014C11.9488 1.32834 10 1.48692e-08 10 1.48692e-08C10 1.48692e-08 8.05124 1.32834 6.04167 2.32014C4.03209 3.31195 0.5 4.54393 0.5 4.54393C0.5 4.54393 0.895833 7.8659 0.895833 9.99446C0.895833 12.123 0.5 15.445 0.5 15.445C0.5 15.445 4.03209 16.677 6.04167 17.6688C7.37825 18.3284 9.03813 19.3738 10 20C10.8146 19.4696 12.13 18.6386 13.3256 17.9959L12.3252 16.2632ZM16.3093 5.49563C15.2543 5.08656 14.0257 4.58369 13.0732 4.11361C12.2914 3.72777 11.5276 3.29969 10.8749 2.91335L10.8749 8.61383L16.3093 5.49563ZM9.12494 2.91342C8.47228 3.29974 7.70853 3.7278 6.92681 4.11361C5.94426 4.59854 4.66775 5.11836 3.59124 5.53412L9.12494 8.70932L9.12494 2.91342ZM2.7562 7.07259C2.83171 8.02424 2.89583 9.09342 2.89583 9.99446C2.89583 11.0156 2.81347 12.2527 2.72553 13.2899L8.15859 10.1725L2.7562 7.07259ZM3.96092 14.5987C4.95085 14.9869 6.05464 15.4449 6.92681 15.8753C7.63127 16.223 8.40485 16.6615 9.12494 17.0927L9.12494 11.6356L3.96092 14.5987ZM17.2831 13.3904C17.192 12.3333 17.1042 11.0488 17.1042 9.99446C17.1042 9.06167 17.1729 7.94868 17.2518 6.97244L11.6749 10.1725L17.2831 13.3904Z'
      />
    </React.Fragment>
  </SvgWrapper>
);

/*
 * /!\ This is a generated file, do not update it or create a similar one by hand.
 * Please refer to the doc in `documentation/ICON_STEPBYSTEP.md` to learn how to deal with icons.
 */
