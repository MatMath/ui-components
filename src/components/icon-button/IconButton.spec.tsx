import * as React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import { IconButton } from './IconButton';
import { CopyPasswordIcon } from '@atoms/icons';

describe('<IconButton>', () => {
  describe('Global render', () => {
    it('should render default', () => {
      const button = mount(<IconButton icon={<CopyPasswordIcon />} />);

      expect(button).toMatchSnapshot();
    });
  });

  describe('Style properties', () => {
    it('should render small size', () => {
      const button = mount(
        <IconButton icon={<CopyPasswordIcon />} size='small' />
      );

      expect(button).toMatchSnapshot();
    });

    it('should render medium size', () => {
      const button = mount(
        <IconButton icon={<CopyPasswordIcon />} size='medium' />
      );

      expect(button).toMatchSnapshot();
    });

    it('should render large size', () => {
      const button = mount(
        <IconButton icon={<CopyPasswordIcon />} size='large' />
      );

      expect(button).toMatchSnapshot();
    });
  });

  describe('Icon render', () => {
    it('should render the specified icon', () => {
      const button = mount(<IconButton icon={<CopyPasswordIcon />} />);

      expect(button.find('svg')).toBeTruthy();
    });
  });

  describe('Interaction', () => {
    it('should execute action on click', () => {
      const button = mount(
        <IconButton
          icon={<CopyPasswordIcon />}
          onClick={() => window.alert('hello')}
        />
      );

      window.alert = jest.fn();
      button.simulate('click');
      expect(window.alert).toHaveBeenCalledTimes(1);
      expect(window.alert).toHaveBeenCalledWith('hello');
    });

    it('should not execute action on click when the button is disabled', () => {
      const button = mount(
        <IconButton
          icon={<CopyPasswordIcon />}
          onClick={() => window.alert('hello')}
          disabled
        />
      );

      window.alert = jest.fn();
      button.simulate('click');
      expect(window.alert).toHaveBeenCalledTimes(0);
    });
  });
});
