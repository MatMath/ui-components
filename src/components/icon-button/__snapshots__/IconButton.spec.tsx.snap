// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton> Global render should render default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20px;
  height: 20px;
  cursor: inherit;
  outline: none;
  border-radius: 4px;
  fill: #0E353D;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  opacity: 1;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.c0:hover {
  fill: #0E353D;
}

.c0:active {
  fill: #0E353D;
}

<Component
  icon={<Unknown />}
>
  <Component
    className="iconButton small"
    nature="ghost"
    size="small"
    theme="light"
  >
    <button
      className="button ghost small light iconButton small"
    >
      <span
        className="content"
      >
        <Component>
          <Component>
            <styled.svg
              svgAnimated={false}
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [
                      [Function],
                    ],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "c0",
                      "rules": Array [
                        "
  display: flex;
  width: ",
                        [Function],
                        ";
  height: ",
                        [Function],
                        ";
  cursor: ",
                        [Function],
                        ";
  outline: none;
  border-radius: 4px;
  fill: ",
                        [Function],
                        ";
  transform: ",
                        [Function],
                        ";
  opacity: ",
                        [Function],
                        ";
  transition: ",
                        [Function],
                        "
  &:hover {
    fill: ",
                        [Function],
                        ";
  }
  &:active {
    fill: ",
                        [Function],
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.svg",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "svg",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                svgAnimated={false}
              >
                <svg
                  className="c0"
                  version="1.1"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.25 9.75V10.75H13.25V12.25H12.25V13.25H10.75V12.25H9.75V10.75H10.75V9.75H12.25Z"
                  />
                   
                  <path
                    clipRule="evenodd"
                    d="M6.5 17.5H16.5V5.50006H6.5V17.5ZM15 7.00006V16H8V7.00006H15Z"
                    fillRule="evenodd"
                  />
                   
                  <path
                    d="M3.5 5.07142H5V4H7.7857V2.5H3.5V5.07142Z"
                  />
                   
                  <path
                    d="M9.21427 2.5H13.5V5.07142H12V4H9.21427V2.5Z"
                  />
                   
                  <path
                    d="M6.5 14.5V13H5V11.9286H3.5V14.5H6.5Z"
                  />
                   
                  <path
                    d="M3.5 11.0714V5.92856H5V11.0714H3.5Z"
                  />
                </svg>
              </StyledComponent>
            </styled.svg>
          </Component>
        </Component>
      </span>
    </button>
  </Component>
</Component>
`;

exports[`<IconButton> Style properties should render large size 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20px;
  height: 20px;
  cursor: inherit;
  outline: none;
  border-radius: 4px;
  fill: #0E353D;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  opacity: 1;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.c0:hover {
  fill: #0E353D;
}

.c0:active {
  fill: #0E353D;
}

<Component
  icon={<Unknown />}
  size="large"
>
  <Component
    className="iconButton large"
    nature="ghost"
    size="large"
    theme="light"
  >
    <button
      className="button ghost large light iconButton large"
    >
      <span
        className="content"
      >
        <Component>
          <Component>
            <styled.svg
              svgAnimated={false}
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [
                      [Function],
                    ],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "c0",
                      "rules": Array [
                        "
  display: flex;
  width: ",
                        [Function],
                        ";
  height: ",
                        [Function],
                        ";
  cursor: ",
                        [Function],
                        ";
  outline: none;
  border-radius: 4px;
  fill: ",
                        [Function],
                        ";
  transform: ",
                        [Function],
                        ";
  opacity: ",
                        [Function],
                        ";
  transition: ",
                        [Function],
                        "
  &:hover {
    fill: ",
                        [Function],
                        ";
  }
  &:active {
    fill: ",
                        [Function],
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.svg",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "svg",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                svgAnimated={false}
              >
                <svg
                  className="c0"
                  version="1.1"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.25 9.75V10.75H13.25V12.25H12.25V13.25H10.75V12.25H9.75V10.75H10.75V9.75H12.25Z"
                  />
                   
                  <path
                    clipRule="evenodd"
                    d="M6.5 17.5H16.5V5.50006H6.5V17.5ZM15 7.00006V16H8V7.00006H15Z"
                    fillRule="evenodd"
                  />
                   
                  <path
                    d="M3.5 5.07142H5V4H7.7857V2.5H3.5V5.07142Z"
                  />
                   
                  <path
                    d="M9.21427 2.5H13.5V5.07142H12V4H9.21427V2.5Z"
                  />
                   
                  <path
                    d="M6.5 14.5V13H5V11.9286H3.5V14.5H6.5Z"
                  />
                   
                  <path
                    d="M3.5 11.0714V5.92856H5V11.0714H3.5Z"
                  />
                </svg>
              </StyledComponent>
            </styled.svg>
          </Component>
        </Component>
      </span>
    </button>
  </Component>
</Component>
`;

exports[`<IconButton> Style properties should render medium size 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20px;
  height: 20px;
  cursor: inherit;
  outline: none;
  border-radius: 4px;
  fill: #0E353D;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  opacity: 1;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.c0:hover {
  fill: #0E353D;
}

.c0:active {
  fill: #0E353D;
}

<Component
  icon={<Unknown />}
  size="medium"
>
  <Component
    className="iconButton medium"
    nature="ghost"
    size="medium"
    theme="light"
  >
    <button
      className="button ghost medium light iconButton medium"
    >
      <span
        className="content"
      >
        <Component>
          <Component>
            <styled.svg
              svgAnimated={false}
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [
                      [Function],
                    ],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "c0",
                      "rules": Array [
                        "
  display: flex;
  width: ",
                        [Function],
                        ";
  height: ",
                        [Function],
                        ";
  cursor: ",
                        [Function],
                        ";
  outline: none;
  border-radius: 4px;
  fill: ",
                        [Function],
                        ";
  transform: ",
                        [Function],
                        ";
  opacity: ",
                        [Function],
                        ";
  transition: ",
                        [Function],
                        "
  &:hover {
    fill: ",
                        [Function],
                        ";
  }
  &:active {
    fill: ",
                        [Function],
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.svg",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "svg",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                svgAnimated={false}
              >
                <svg
                  className="c0"
                  version="1.1"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.25 9.75V10.75H13.25V12.25H12.25V13.25H10.75V12.25H9.75V10.75H10.75V9.75H12.25Z"
                  />
                   
                  <path
                    clipRule="evenodd"
                    d="M6.5 17.5H16.5V5.50006H6.5V17.5ZM15 7.00006V16H8V7.00006H15Z"
                    fillRule="evenodd"
                  />
                   
                  <path
                    d="M3.5 5.07142H5V4H7.7857V2.5H3.5V5.07142Z"
                  />
                   
                  <path
                    d="M9.21427 2.5H13.5V5.07142H12V4H9.21427V2.5Z"
                  />
                   
                  <path
                    d="M6.5 14.5V13H5V11.9286H3.5V14.5H6.5Z"
                  />
                   
                  <path
                    d="M3.5 11.0714V5.92856H5V11.0714H3.5Z"
                  />
                </svg>
              </StyledComponent>
            </styled.svg>
          </Component>
        </Component>
      </span>
    </button>
  </Component>
</Component>
`;

exports[`<IconButton> Style properties should render small size 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20px;
  height: 20px;
  cursor: inherit;
  outline: none;
  border-radius: 4px;
  fill: #0E353D;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  opacity: 1;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.c0:hover {
  fill: #0E353D;
}

.c0:active {
  fill: #0E353D;
}

<Component
  icon={<Unknown />}
  size="small"
>
  <Component
    className="iconButton small"
    nature="ghost"
    size="small"
    theme="light"
  >
    <button
      className="button ghost small light iconButton small"
    >
      <span
        className="content"
      >
        <Component>
          <Component>
            <styled.svg
              svgAnimated={false}
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [
                      [Function],
                    ],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "c0",
                      "rules": Array [
                        "
  display: flex;
  width: ",
                        [Function],
                        ";
  height: ",
                        [Function],
                        ";
  cursor: ",
                        [Function],
                        ";
  outline: none;
  border-radius: 4px;
  fill: ",
                        [Function],
                        ";
  transform: ",
                        [Function],
                        ";
  opacity: ",
                        [Function],
                        ";
  transition: ",
                        [Function],
                        "
  &:hover {
    fill: ",
                        [Function],
                        ";
  }
  &:active {
    fill: ",
                        [Function],
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.svg",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "svg",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                svgAnimated={false}
              >
                <svg
                  className="c0"
                  version="1.1"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.25 9.75V10.75H13.25V12.25H12.25V13.25H10.75V12.25H9.75V10.75H10.75V9.75H12.25Z"
                  />
                   
                  <path
                    clipRule="evenodd"
                    d="M6.5 17.5H16.5V5.50006H6.5V17.5ZM15 7.00006V16H8V7.00006H15Z"
                    fillRule="evenodd"
                  />
                   
                  <path
                    d="M3.5 5.07142H5V4H7.7857V2.5H3.5V5.07142Z"
                  />
                   
                  <path
                    d="M9.21427 2.5H13.5V5.07142H12V4H9.21427V2.5Z"
                  />
                   
                  <path
                    d="M6.5 14.5V13H5V11.9286H3.5V14.5H6.5Z"
                  />
                   
                  <path
                    d="M3.5 11.0714V5.92856H5V11.0714H3.5Z"
                  />
                </svg>
              </StyledComponent>
            </styled.svg>
          </Component>
        </Component>
      </span>
    </button>
  </Component>
</Component>
`;
